azext_iot/iothub/providers/base.py

```diff
@@ -29,3 +31,17 @@ def __init__(self, cmd, hub_name, rg, login=None, auth_type_dataplane=None):
 
     def get_sdk(self, sdk_type):
         return self.resolver.get_sdk(sdk_type)
+
+
+class IoTHubResourceProvider(object):
```

Comment:
since this no longer does weird stuff with getting the client, we could just add this into the normal iothub provider

Reply:
Note that if we do, we should make sure that the discovery doesnt get fetch target for message route/endpoints to reduce run time